packages/next/src/utilities/serverFunctions/login.ts

```diff
@@ -0,0 +1,104 @@
+'use server'
+
+import type { CollectionSlug } from 'payload'
+
+import { cookies as getCookies } from 'next/headers.js'
+import { generatePayloadCookie, getPayload } from 'payload'
+
+export async function login({
+  collection,
+  config,
+  email,
+  password,
+  username,
+}: {
+  collection: CollectionSlug
+  config: any
+  email?: string
+  password: string
+  username?: string
```

Comment:
Need to update the types here to restrict that _either_ an email _or_ a username is accepted in the args. You should not be allowed to send both.

Reply:
Updated